frontend/public/extend/devconsole/components/deployment-config/DeploymentConfig.tsx

```diff
@@ -0,0 +1,81 @@
+/*eslint-disable no-unused-vars, no-undef */
+import * as React from 'react';
+import { Checkbox } from './../../../../../public/components/checkbox';
+import { EnvironmentPage } from './../../../../../public/components/environment';
+
+interface NameValueType {
+  name: string;
+  value: string;
+}
+
+interface NameValueFormType {
+  name: string;
+  valueForm: {
+    configMapKeyRef: {
+      key: string;
+      name: string
+    }
+  }
+}
+
+interface BuildConfigProps {
+  onNewImageAvailableChange: React.ReactEventHandler<HTMLInputElement>;
+  onDeploymentConfigChange: React.ReactEventHandler<HTMLInputElement>;
+  onEnviromentVariableChange: (obj: NameValueType | NameValueFormType) => void;
```

Comment:
The callback contains an array of pairs.
```suggestion
  onEnviromentVariableChange: (envPairs: (NameValueType | NameValueFormType)[]) => void;
```